eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTests.java

```diff
@@ -0,0 +1,356 @@
+        PowerMockito.when(instanceInfo.getStatus())
+                .thenReturn(InstanceInfo.InstanceStatus.UP);
+        PowerMockito.when(instanceInfo.getActionType())
+                .thenReturn(InstanceInfo.ActionType.ADDED);
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        ArrayList<Application> applicationArrayList = new ArrayList<>();
+        applicationArrayList.add(application);
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList);
+
+        Assert.assertEquals(1, EurekaEntityFunctions
+                .mergeApplications(applications, applications).size());
+    }
+
+    @Test
+    public void testMergeApplications2() {
+        InstanceInfo instanceInfo1 = PowerMockito.mock(InstanceInfo.class);
+        PowerMockito.when(instanceInfo1.getId()).thenReturn("foo");
+        PowerMockito.when(instanceInfo1.getStatus())
+                .thenReturn(InstanceInfo.InstanceStatus.UP);
+        PowerMockito.when(instanceInfo1.getActionType())
+                .thenReturn(InstanceInfo.ActionType.ADDED);
+        Application application1 = new Application("foo");
+        application1.addInstance(instanceInfo1);
+
+        ArrayList<Application> applicationArrayList1 = new ArrayList<>();
+        applicationArrayList1.add(application1);
+        Applications applications1 = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList1);
+
+        InstanceInfo instanceInfo2 = PowerMockito.mock(InstanceInfo.class);
+        PowerMockito.when(instanceInfo2.getId()).thenReturn("bar");
+        PowerMockito.when(instanceInfo2.getStatus())
+                .thenReturn(InstanceInfo.InstanceStatus.UP);
+        PowerMockito.when(instanceInfo2.getActionType())
+                .thenReturn(InstanceInfo.ActionType.ADDED);
+        Application application2 = new Application("bar");
+        application2.addInstance(instanceInfo2);
+
+        ArrayList<Application> applicationArrayList2 = new ArrayList<>();
+        applicationArrayList2.add(application2);
+        Applications applications2 = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList2);
+
+        Assert.assertEquals(2, EurekaEntityFunctions
+                .mergeApplications(applications1, applications2).size());
+    }
+
+    @Test
+    public void testMergeApplication1() {
+        InstanceInfo instanceInfo = PowerMockito.mock(InstanceInfo.class);
+        PowerMockito.when(instanceInfo.getId()).thenReturn("foo");
+        PowerMockito.when(instanceInfo.getActionType())
+                .thenReturn(InstanceInfo.ActionType.ADDED);
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        Assert.assertEquals(application.getInstances(), EurekaEntityFunctions
+                .mergeApplication(application, application).getInstances());
```

Comment:
Change line break if possible. It looks like this might require two breaks.

Reply:
done